js_modules/dagit/packages/core/src/app/ExecutionSessionStorage.tsx

```diff
@@ -105,26 +109,10 @@ export function applyCreateSession(
   };
 }
 
-// StorageProvider component that vends `IStorageData` via a render prop
-
 type StorageHook = [IStorageData, (data: IStorageData) => void];
 
-let _data: IStorageData | null = null;
-let _dataNamespace = '';
-
-function getKey(namespace: string) {
-  return `dagit.v2.${namespace}`;
-}
-
-function getStorageDataForNamespace(namespace: string, initial: Partial<IExecutionSession> = {}) {
-  if (_data && _dataNamespace === namespace) {
-    return _data;
-  }
-
-  let data: IStorageData = Object.assign(
-    {sessions: {}, current: ''},
-    getJSONForKey(getKey(namespace)),
-  );
+const validate = (initial: Partial<IExecutionSession> = {}) => (json: any): IStorageData => {
```

Comment:
I'd missed that there were two arrows here, might be nice to call this `buildValidator` or something like that to make it more explicit that it's a factory method